### PR TITLE
BREAKING(path): split up glob into multiple files

### PR DESCRIPTION
**Pull request by lino-levan**
Closes https://github.com/denoland/deno_std/issues/3600

Please take a careful look, I might've missed something.

### CODE REVIEW

**Comment by lino-levan** [comment]
Looks like several flaky tests :/

Any idea why these are happening now @iuioiua?

**Comment by iuioiua** [comment]
I don’t know the fix. I’ve asked Leo if he has any ideas.

File: path/glob.ts [hunk 1]
```diff
@@ -1,28 +1,26 @@
...
 
-export type GlobToRegExpOptions = GlobOptions & {
-  os?: OSType;
-};
+/**
+ * @deprecated (will be removed in 0.215.0) Import from "std/path/windows/is_glob.ts"
+ *
+ * Test whether the given string is a glob
+ */
+export const isGlob = _isGlob;
```
Comment (iuioiua) [inline path/glob.ts:18]:
```suggestion
export {
  /**
   * @deprecated (will be removed in 0.215.0) Import from "std/path/windows/is_glob.ts"
   *
   * Test whether the given string is a glob
   */
  isGlob,
} from "./is_glob.ts";
```
I think this approach is cleaner. Ditto for the other instances.

**Review by iuioiua — CHANGES_REQUESTED** [review]

File: path/glob.ts (hunk 1 above)
Comment (iuioiua) [reply path/glob.ts:18]: Yep 👍🏾

**Review by iuioiua — COMMENTED** [review]
None of these glob functions are platform-specific. Their implementations are identical and duplicated across both platforms. They should be split up at the top level of `std/path` but not in platform-specific folders. Am I missing something here?

**Comment by iuioiua** [comment]
Oh, I see they use different `normalize()` functions... Nevermind. Please allow a little time for me to comb through this.

**Review by kt3k — APPROVED** [review]
LGTM

**Review by iuioiua — APPROVED** [review]
LGTM. However, there's some duplicated code between the two functions. E.g. `normalizeGlob()` are essentially the same on both platforms, except that they rely on different underlying `normalize()` and `SEP_PATTERN` symbols. Let's discuss options for making this better in an issue.

**Comment by kt3k** [comment]
@lino-levan Thank you for your contribution!

**Pull request merged**